File: BloodBankBackend/src/main/java/com/itvedant/bloodbankmanagement/service/BloodRequestService.java
package com.itvedant.bloodbankmanagement.service;

import com.itvedant.bloodbankmanagement.entity.BloodRequest;
import java.util.List;

public interface BloodRequestService {
    List<BloodRequest> getAllRequests();
    BloodRequest getRequestById(Long id);
    BloodRequest createRequest(BloodRequest request);
    BloodRequest updateRequest(Long id, BloodRequest requestDetails);
    void deleteRequest(Long id);
    
    BloodRequest fulfillRequest(Long id);
}

File: BloodBankFrontend/bloodbank/src/services/BloodRequestService.js
import authAxios from "./authAxios";

const API_URL = "/api/bloodrequests";

class BloodRequestService {
  getAllRequests() {
    return authAxios.get(API_URL);
  }

  getRequestById(id) {
    return authAxios.get(`${API_URL}/${id}`);
  }

  createRequest(request) {
    return authAxios.post(API_URL, request);
  }

  updateRequest(id, request) {
    return authAxios.put(`${API_URL}/${id}`, request);
  }

  deleteRequest(id) {
    return authAxios.delete(`${API_URL}/${id}`);
  }

   fulfillRequest(id) {
        return authAxios.post(`${API_URL}/${id}/fulfill`);
    }
}

const bloodRequestService = new BloodRequestService();
export default bloodRequestService;
